### PR TITLE
Periodically emit consul_version metric instead of just once on startup

### PR DESCRIPTION
**Pull request by analogue**
### Description
See https://hashicorp.atlassian.net/browse/NET-1643. 

The `consul_version` metric is currently emitted once on startup via `metrics.SetGaugeWithLabels(...)`.  

When Prometheus metrics are enabled by setting the `telemetry -> prometheus_retention_time` config value to a non-zero value, any metric which hasn't been emitted in the last `prometheus_retention_time` seconds (using seconds as unit of time as an example) will get discarded.

From a UX perspective, discarded means that `curl`'ing the Prometheus endpoint no longer includes the `consul_version` metric. This is surprising because users depend on the `consul_version` metrics always being there so they can detect/confirm the version has changed during deployments/upgrades.

**Option 1**
Spin up a dedicated goroutine to periodically emit `consul_version` every `prometheus_retention_time - 1` seconds so that it never meets the criteria to be discarded.

**Option 2**
Piggyback on an existing set of periodically emitted metrics (goroutine that emits on regular intervals) and just add `consul_version` to that set of metrics instead.

**Decision**
We're going with option 2 here as the "simplest thing possible" since `Usage Metrics` are already emitted on an interval and not tied to a specific feature (e.g. mesh) that could be disabled via configuration.

### Testing & Reproduction steps

TODO

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -860,12 +860,6 @@ func (a *Agent) Start(ctx context.Context) error {
 		go m.Monitor(&lib.StopChannelContext{StopCh: a.shutdownCh})
 	}
 
-	// consul version metric with labels
-	metrics.SetGaugeWithLabels([]string{"version"}, 1, []metrics.Label{
-		{Name: "version", Value: a.config.VersionWithMetadata()},
-		{Name: "pre_release", Value: a.config.VersionPrerelease},
```
Comment (analogue) [inline agent/agent.go:866]: You probably want to emit the "pre_release" label too to stay consistent with the existing implementation.

File: agent/agent.go [hunk 2]
```diff
@@ -860,12 +860,6 @@ func (a *Agent) Start(ctx context.Context) error {
 		go m.Monitor(&lib.StopChannelContext{StopCh: a.shutdownCh})
 	}
 
-	// consul version metric with labels
-	metrics.SetGaugeWithLabels([]string{"version"}, 1, []metrics.Label{
-		{Name: "version", Value: a.config.VersionWithMetadata()},
```
Comment (analogue) [inline agent/agent.go:865]: Have the `emitVersionUsage()` emit the same value `VerstionWithMetadata()` instead of `GetHumanVersion()`?

Comment (skpratt) [reply agent/agent.go:865]: VersionWithMetadata looks like it just concats Version and VersionMetadata [here](https://github.com/hashicorp/consul/blob/255aa158db8ccbfd445205654e153742b127116d/agent/config/runtime.go#L1738), which get input from the version module [here](https://github.com/hashicorp/consul/blob/255aa158db8ccbfd445205654e153742b127116d/agent/config/default.go#L268), so this should be doable.

File: agent/agent.go (hunk 1 above)
Comment (20sr20) [reply agent/agent.go:866]: I have noticed that for OSS/ent version, pre-release is empty.  Only for dev , pre-release is showed up as `dev`, which is understanble, given that the dev version is before release. I din't understand the logic behind putting "pre_release" label here in the metrics.  Customer will not have prerelease version, they would just want to see their Consul version (either OSS or ent) in their prometheus metrics.  Without `pre release` labelling, I got the output version="1.17.0-dev" as below.  Please let me know your opinion.

# HELP consul_version - Represents the Consul version.
# TYPE consul_version gauge
consul_version 0
consul_version{datacenter="dc1",version="1.17.0-dev"} 1

Comment (20sr20) [reply agent/agent.go:866]: We can ignore my above comment.  I will see how I can align the format with the existing implementation - `{Name: "pre_release", Value: a.config.VersionPrerelease},`

**Comment by analogue** [comment]
@20sr20 Could you fix the failing tests @ https://github.com/hashicorp/consul/actions/runs/6043014854/job/16399304426?pr=18622 ? The error string across a number of failures is

`   usagemetrics_oss_test.go:1471: found unexpected gauge key:      consul.usage.test.version;version=1.17.0;pre_release=dev`

**Comment by 20sr20** [comment]
@analogue I have fixed that. Please review again.  Thanks so much for looking into this

**Comment by analogue** [comment]
Closing and replacing with a PR on ENT that will be backported to CE: https://github.com/hashicorp/consul-enterprise/pull/6876

**Pull request closed without merging**